panoptes_aggregation/extractors/utilities.py

```diff
@@ -11,3 +11,57 @@ def annotation_by_task(classification_in):
         ann_by_task[annotation['task']].append(annotation)
     classification['annotations'] = dict(ann_by_task)
     return classification
+
+
+def pluck_fields(classification, pluck_keys):
+    '''
+    Main function that is called by the pluck_extractor.
+    Functions similarly to the PluckFieldExtractor
+    on Caesar, where given a mapping between a key and
+    location of the data on the classification JSON,
+    this extractor returns a dictionary with each extracted
+    value given by the corresponding key.
+
+    Input
+    -----
+    classification : dict
+        A dictionary containing an `annotations` key that is a list of
+        panoptes annotations
+
+    pluck_keys : string
+        String with a list-like formatting that maps the output key to
+        the location of the value in the classification dict. Must begin
+        with a `[` and end with `]`, with each key map separated by a `,`.
+        Each entry should be in the form key:location (see examples below).
```

Comment:
I am not convinced this is the best format for `pluck_keys`.  If this is instead passed in as a dictionary on the URL you can run: 

```python
import ast
pluck_key_dict = ast.literal_eval(pluck_keys)

for key, value in pluck_key_dict.items():
   ...
```

Less code and also matches the same way subtask information is passed into extractors and reducers for drawing tasks.

Note: My main concern is string parsing can get quite messy and lead to weird edge cases when badly formatted strings are passed into them.  `ast.literal_eval` will at some level protect from arbitrary code injection https://docs.python.org/3/library/ast.html#ast.literal_eval

Reply:
I made the necessary changes to the `pluck_fields` function

Reply:
🤦 I edited this comment but forgot to hit save before submitting the review. After I first wrote this I thought of a better way to do this that is more in line with the other code in the repo.  I will put some inline comments in the new code.